vendor/k8s.io/kube-openapi/pkg/handler/handler.go

```diff
@@ -74,28 +79,64 @@ func computeETag(data []byte) string {
 	return fmt.Sprintf("\"%X\"", sha512.Sum512(data))
 }
 
+type specFetcherFunc func() (spec *spec.Swagger, err error)
+
 // NewOpenAPIService builds an OpenAPIService starting with the given spec.
-func NewOpenAPIService(spec *spec.Swagger) (*OpenAPIService, error) {
+func NewOpenAPIService(spec *spec.Swagger, fetcher specFetcherFunc) (*OpenAPIService, error) {
 	o := &OpenAPIService{}
+	// TODO(DangerOnTheRanger): is it safe to remove this UpdateSpec call?
```

Comment:
@apelisse I think you had some thoughts around whether we should generate the OpenAPI on startup?

One thing I'll add is that the aggregation openapi controller will fetch the `/openapi/v2` from the builtins and CRDs on start up, so deferring the aggregation should only benefit the aggregated api server.

Reply:
> the aggregation openapi controller will fetch the `/openapi/v2` from the builtins and CRDs on start up

Do you know why it does that?

Reply:
It's a convention to populate the cache before registering the `openapi/v2` handler.

Reply:
That doesn't sound required then, is there a technical reason?

Reply:
I don't think there is if I understand this area of the code correctly. I'll go ahead and remove it unless someone else raises objections ~~and/or something horribly breaks in the process~~.